game-app/game-core/src/main/java/games/strategy/engine/data/TechnologyFrontier.java

```diff
@@ -56,12 +57,28 @@ public void removeAdvance(final TechAdvance t) {
     getData().getTechTracker().clearCache();
   }
 
-  public TechAdvance getAdvanceByProperty(final String property) {
-    return techs.stream().filter(ta -> ta.getProperty().equals(property)).findAny().orElse(null);
+  /**
+   * Returns {@link TechAdvance} by search first via {@link
+   * TechnologyFrontier#getAdvanceByProperty(String)} and then by {@link
+   * TechnologyFrontier#getAdvanceByName(String)}.
+   *
+   * @param propertyOrName Search string
+   * @return {@link Optional} of {@link TechAdvance}
+   */
```

Comment:
I think this javadoc would be more helpful if it focused on:
- what is the difference of name vs property and where is each one found
- what are the conditions for returning an empty

I find the {@link} with very long names difficult to read as well. It really seems to detract. It's a bit interesting too that documentation on public methods links to private methods. For both reasons, I'd strongly consider removing the {@link} texts.

FWIW, as soon as we can migrate to javadoc in 'markdown' sytnax, I'll be pushing for that hard. I think we are free to use backticks right now instead of using {@link for all types. EG: 

> `Optional` 

instead of

`{@link Optional`